oracle/src/confirmRelay.js

```diff
@@ -0,0 +1,155 @@
+require('../env')
+const path = require('path')
+const { connectWatcherToQueue, connection } = require('./services/amqpClient')
+const logger = require('./services/logger')
+const rpcUrlsManager = require('./services/getRpcUrlsManager')
+const { getEventsFromTx } = require('./tx/web3')
+const { checkHTTPS, watchdog } = require('./utils/utils')
+const { EXIT_CODES } = require('./utils/constants')
+
+if (process.argv.length < 5) {
+  logger.error('Please check the number of arguments, transaction hash is not present')
+  process.exit(EXIT_CODES.GENERAL_ERROR)
+}
+
+const config = require(path.join('../config/', process.argv[2]))
+const txHash = process.argv[4]
+
+const processSignatureRequests = require('./events/processSignatureRequests')(config)
+const processCollectedSignatures = require('./events/processCollectedSignatures')(config)
+const processAffirmationRequests = require('./events/processAffirmationRequests')(config)
+const processTransfers = require('./events/processTransfers')(config)
+const processAMBSignatureRequests = require('./events/processAMBSignatureRequests')(config)
+const processAMBCollectedSignatures = require('./events/processAMBCollectedSignatures')(config)
+const processAMBAffirmationRequests = require('./events/processAMBAffirmationRequests')(config)
+
+const { getTokensState } = require('./utils/tokenState')
+
+const web3Instance = config.web3
+const bridgeContract = new web3Instance.eth.Contract(config.bridgeAbi, config.bridgeContractAddress)
+let { eventContractAddress } = config
+let eventContract = new web3Instance.eth.Contract(config.eventAbi, eventContractAddress)
+
+async function initialize() {
+  try {
+    const checkHttps = checkHTTPS(process.env.ORACLE_ALLOW_HTTP_FOR_RPC, logger)
+
+    rpcUrlsManager.homeUrls.forEach(checkHttps('home'))
+    rpcUrlsManager.foreignUrls.forEach(checkHttps('foreign'))
+
+    connectWatcherToQueue({
+      queueName: config.queue,
+      workerQueue: config.workerQueue,
+      cb: runMain
+    })
+  } catch (e) {
+    logger.error(e)
+    process.exit(EXIT_CODES.GENERAL_ERROR)
+  }
+}
+
+async function runMain({ sendToQueue, sendToWorker }) {
+  try {
+    if (connection.isConnected()) {
+      if (config.maxProcessingTime) {
+        await watchdog(() => main({ sendToQueue, sendToWorker, txHash }), config.maxProcessingTime, () => {
+          logger.fatal('Max processing time reached')
+          process.exit(EXIT_CODES.MAX_TIME_REACHED)
+        })
+      } else {
+        await main({ sendToQueue, sendToWorker, txHash })
+      }
+    } else {
+      setTimeout(() => {
+        runMain({ sendToQueue, sendToWorker })
+      }, config.pollingInterval)
+    }
+  } catch (e) {
+    logger.error(e)
+  }
+}
+
+function processEvents(events) {
+  switch (config.id) {
+    case 'native-erc-signature-request':
```

Comment:
Please clarify why do we need to process Signature Requests and Collected Signatures events here?

Reply:
I decided to keep almost all events and processing logic from `oracle/src/watcher.js`. So it should be possible to process any individual transaction using new script, if it contains some unprocessed events (including Signature requests / Collected signatures events).

Reply:
I understood this. And my worry is that we don't lose usability by trying to achieve universality.  Please confirm that for the bad case the person who runs this functionality will get a descriptive reason why the action cannot be completed (e.g. that confirmation from this validator has been already registered in the contract).

Reply:
This reasons are present in the handling logic in `oracle/src/events/*`, so everything should be ok.
For example, for some already handled transaction I am getting the following message while trying to process `CollectedSignatures` event:
`[1589897350889] INFO  (watcher-erc-native-collected-signatures): Validator not responsible for relaying CollectedSignatures 0xf0d89e4cd07981d528c2915b4f7ba3878b0aa2b849dba21f952fc815d1bca5e5`

Reply:
thanks for explanation